src/main/java/com/spartronics4915/frc2020/subsystems/LED.java

```diff
@@ -0,0 +1,100 @@
+package com.spartronics4915.frc2020.subsystems;
+
+import com.spartronics4915.lib.util.Logger;
+import com.spartronics4915.lib.subsystems.SpartronicsSubsystem;
+
+import edu.wpi.first.wpilibj.SerialPort;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+/**
+ * The subsystem that controls the LED.
+ * Please see the README.md in the Arduino_Bling directory for details.
+ */
+public class LED extends SpartronicsSubsystem
+{
+    private SerialPort mBling;
+
+    private static LED sInstance = null;
+    private static BlingState mBlingState;
+
+    public static LED getInstance()
+    {
+        if (sInstance == null)
+        {
+            sInstance = new LED();
+            mBlingState = BlingState.OFF;
+        }
+        return sInstance;
+    }
+
+    private LED()
+    {
+        try
+        {
+            mBling = new SerialPort(9600, SerialPort.Port.kUSB);
+            logInitialized(true);
+        }
+        catch (Exception e)
+        {
+            logException("LED: Couldn't initialize SerialPort", e);
+            logInitialized(false);
+        }
+    }
+
+    // TODO: validate desired periodic() function for LED subsystem
+    @Override
+    public void periodic()
+    {
+        SmartDashboard.putString("LED state:", mBlingState.toString());
+    }
+
+    /**
+     * This enum is giving the possible styles we can have the Arduino express.
+     */
+    public enum BlingState
```

Comment:
What I went over above - instead of states, our robot code is using specific methods called by commands (a good example - [Climber.java](https://github.com/Spartronics4915/2020-InfiniteRecharge/blob/master/src/main/java/com/spartronics4915/frc2020/subsystems/Climber.java)), with the CommandScheduler ensuring only one runs at a time.

Reply:
Note, we don't have the same restriction of running one command at a time for LED system. Animation design problem, which is why I introduced the states (which basically sends an animation code to Arduino), is to ensure we don't frequently switch from one animation to another as it will just result in a mess of blinking lights.

Reply:
That seems like it would be better handled by the `isFinished` and `end` Command methods. If `isFinished` is set to `false`, the command will persist (even with an empty `execute` loop) until another requiring the same subsystem is scheduled. The `end` method can send a clear command to the Arduino.

Reply:
There is no value in keeping 'SetBlingStateCommand' running continuously -- running empty execute still requires the scheduler to check, which uses system resources. From programmers perspective, it is a no-op, as they they still need to specify when it is time to change the pattern.

Reply:
That makes sense - I've merged this PR w/o changing the state structure. If it gives us problems we can always change it after testing...

Reply:
Sorry I missed Wednesday meeting -- I should be in tomorrow as I am feeling better. We'll go over the LED system's design philosophy, including reuse and required contract w/ the Arduino code so you can assess how to integrate command passing when bling states are defined.